feconf.py

```diff
@@ -235,6 +237,11 @@ def get_empty_ratings():
     },
 }
 
+# Prefixes for activity ids to denote the type of activity. These values should
+# not be changed.
+ACTIVITY_ID_PREFIX_EXPLORATION = 'e:'
```

Comment:
I don't think I like the single string approach. It seems way too error-prone. We will have places where not having the prefix is okay and other places when it's not. That just seems messy to me. We really ought to make this stricter. At this point I think a domain object representing an activity type would be rather reasonable. Converting this to/from a dict and having a corresponding frontend domain object would certainly simplify the pipelining. We would have stricter type checking (properties ``is_exploration``, ``is_collection``). The new storage model you introduced can store both the activity ID and its type (would use a repeated JSON property rather than a repeated string property, since it would store a list of dicts).

I'm rather strongly against the use of strings in this way. Maybe we should discuss it more?